src/edu/hillel/Lesson8/ArrayValueCalculator.java

```diff
@@ -0,0 +1,38 @@
+package edu.hillel.Lesson8;
+
+import java.util.ArrayList;
+
+public class ArrayValueCalculator {
+    final static private int ARRAY_DIMENSION_THRESHOLD=4;
+
+    //first method throws ArraySizeException,ArrayDataException.
+    // But I try to add reaction on another global exception - and now method throws global Exception because my exceptions extends of Exception
+    public int doCalc(String[][] arr) throws Exception {
+        if (arr.length>ARRAY_DIMENSION_THRESHOLD){
+            throw new ArraySizeException("Dimension of array "+arr.length+" is over limit="+ARRAY_DIMENSION_THRESHOLD);
+        } else if (arr.length==0) {
+            throw new Exception("Input array is empty!!");
+
+        }
+        for (int i=0;i<arr.length;i++){
+            if (arr[i].length>ARRAY_DIMENSION_THRESHOLD){
+                throw new ArraySizeException("Dimension of line "+i+" in array "+arr[i].length+" is over limit="+ARRAY_DIMENSION_THRESHOLD);
+            }
+        }
+        int totalSum=0;
+        for(int i=0;i< arr.length;i++){
+            for(int j=0;j<arr[i].length;j++){
+                try{
+                    totalSum+=Integer.parseInt(arr[i][j]);
+                }
+                catch (NumberFormatException e){
+                    throw new ArrayDataException("Data in row "+i+" and column "+j+" is \""+arr[i][j]+"\" and cannot be parsed to int!");
```

Comment:
You can overload one more constructor in ArrayDataException like:
public ArraySizeException(String message, Throwable e) {
        super(message, e);
    }

And then put here not only message but NumberFormatException just do not loosing original stacktrace

Reply:
overload all constructors in ArraySizeException and ArrayDataException,
this allow us to get full stack trace and print it to console. 
First I didn't do so because I only wanted to show catching specific exception.
Now I print all stack traces in case of exception.